sound/soc/intel/boards/sof_sdw.c

```diff
@@ -538,10 +469,19 @@ static bool is_unique_device(const struct snd_soc_acpi_link_adr *link,
 static int create_codec_dai_name(struct device *dev,
 				 const struct snd_soc_acpi_link_adr *link,
 				 struct snd_soc_dai_link_component *codec,
-				 int offset)
+				 int offset,
+				 struct snd_soc_codec_conf *codec_conf,
+				 int codec_count,
+				 int *codec_conf_index)
 {
 	int i;
 
+	/* sanity check */
+	if (*codec_conf_index >= codec_count) {
+		dev_err(dev, "codec_conf invalid config1\n");
```

Comment:
is this a preliminary version of the error message? Wouldn't we want to use "error:" as a prefix here too? And maybe it's only me, but "config1" and "config2" don't seem very descriptive to me :-) In fact we probably want to check in advance that we aren't overwriting beyond the array in the loop, check for `*codec_conf_index + link->num_adr <= codec_count` or similar?

Reply:
yes this was just temporary. The suggestion is good, thanks